fs-integration-test-suite/accelerator-tests/is-tests/consent-management/src/test/groovy/org/wso2/financial/services/accelerator/is/test/consent/management/ConsentOperationsFlowTests/ConsentFlowTypesValidationTest.groovy

```diff
@@ -0,0 +1,229 @@
+import org.wso2.financial.services.accelerator.test.framework.constant.AccountsRequestPayloads
+import org.wso2.financial.services.accelerator.test.framework.constant.ConnectorTestConstants
+import org.wso2.financial.services.accelerator.test.framework.constant.PageObjects
+import org.wso2.financial.services.accelerator.test.framework.constant.PaymentRequestPayloads
+import org.wso2.financial.services.accelerator.test.framework.request_builder.AuthorisationBuilder
+import org.wso2.financial.services.accelerator.test.framework.utility.ConsentMgtTestUtils
+import org.wso2.financial.services.accelerator.test.framework.utility.TestUtil
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Consent Flow Types Validation Test.
+ * This test class is to validate the consent types supported by the IS connector. It will enable consent flow
+ * types per scope and test whether the correct flow is executed. In this step accounts scope will follow a
+ * scope based consent flow and payments scope will follow a pre initiated consent flow.
+ *
+ * Before running the tests, make sure to enable the consent flow types in the deployment.toml as below.
+ * [financial_services.consent.pre_initiated]
+ * scopes=["payments"]
+
+ * [financial_services.consent.scope_based]
+ * scopes=["accounts"]
+ *
+ * [financial_services.identity]
+ * append_consent_id_to_token_introspect_response=true
+ */
+class ConsentFlowTypesValidationTest extends FSConnectorTest {
+
+    // Test scope based consent flow for accounts scope
+    @Test
+    void "Authorise consent without a pre initiation step for accounts scope"() {
+
+        consentId = ""
+
+        AuthorisationBuilder acceleratorAuthorisationBuilder = new AuthorisationBuilder()
+        String authoriseUrl = acceleratorAuthorisationBuilder.getAuthorizationRequestWithoutIntentId(
+                configuration.getAppInfoClientID(), [ConnectorTestConstants.ApiScope.ACCOUNTS],
+                true).toURI().toString()
+
+        automation = getBrowserAutomation(ConnectorTestConstants.DEFAULT_DELAY)
+                .addStep(new BasicAuthAutomationStep(authoriseUrl))
+                .addStep { driver, context ->
+                    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS)
+
+                    WebDriverWait wait = new WebDriverWait(driver, 10)
+                    if ((driver.findElements(By.xpath(PageObjects.CHK_SALARY_SAVER_ACC))).displayed) {
+                        driver.findElement(By.xpath(PageObjects.CHK_SALARY_SAVER_ACC)).click()
+                    }
+
+                    WebElement btnApprove = wait.until(
+                            ExpectedConditions.elementToBeClickable(By.xpath(PageObjects.BTN_APPROVE)))
+                    btnApprove.click()
+                }
+                .addStep(new WaitForRedirectAutomationStep())
+                .execute()
+
+        // Get Code From URL
+        code = TestUtil.getHybridCodeFromUrl(automation.currentUrl.get())
```

Comment:
This test case doesn't verify any response. Shall we add an assertion to verify the code is not null.